src/libraries/System.Net.Primitives/src/System/Net/SocketException.cs

```diff
@@ -30,13 +31,18 @@ public SocketException(int errorCode) : this((SocketError)errorCode)
             // but that's the least bad option right now.
         }
 
+        public SocketException(int errorCode, EndPoint? endPoint) : this((SocketError)errorCode)
```

Comment:
I think `SocketException(int errorCode, string message)` would be indeed more future-proof, and feels like missing API regardless of the current issue, while storing an IPEndpoint feels like an arbitrary decision.

>  However, RemoteEndPoint.ToString() would always allocate but I'm not sure if that matters that much for failures.

Exception throwing is already very an expensive, and there is a high chance that someone somewhere would log the message anyways, so I would just pass that string in every case.